### DISCUSSION TITLE
[Question] How to build to support nginx proxied version

### DISCUSSION THREAD

**Discussion by jacroe** [tamland/airsonic-refix#30] [General]
I'm already having to do a custom build to make the project work on ARM. So I might as well also change what's necessary to make it support being behind an nginx proxy, as mentioned in #2.

What do I need to change to make the app available at, say, http://example.com/music/ ?

**Comment by tamland** [comment]
To get the build correct it should be sufficient to change [publicPath](https://cli.vuejs.org/config/#publicpath) to `music/` (Not tested tho). Might need some additional nginx magic, I'm not sure.

**Reply by CustardFlan** [reply]
Indeed, it works, compiled an tested with nginx
vue.config.js:
module.exports = {
      publicPath: '/airsonic2/',
  devServer: {
    disableHostCheck: true,
    overlay: {
      errors: false
    }
  }

}

**Reply by polhaghverdian** [reply]
May I ask how your nginx.conf looks like?

**Comment by gjedeer** [comment]
@tamland the link you sent says:

> The value can also be set to an empty string ('') or a relative path (./) so that all assets are linked using relative paths. This allows the built bundle to be deployed under any public path, or used in a file system based environment like a Cordova hybrid app.

Would it be possible to build it with an empty path by default?

**Reply by tamland** [reply]
Below it says:
> Relative publicPath has some limitations and should be avoided when: You are using HTML5 history.pushState routing;

No sure what the "some limitations" are though..

**Comment by ToxicFrog** [comment]
Based on personal testing: with relative paths, it functions, but the browser URL rapidly becomes wrong (it loses the part it's relative to, so the app continues functioning since it's all JS and XHR but the browser history is invalid reloading the page completely will fail) and some images fail to load.

**Comment by c-markwell** [comment]
I changed the base path by modifying the "base" value in [/vite.config.mjs](https://github.com/tamland/airsonic-refix/blob/f5a14e3f95814c13c95deb17c061635c9247401c/vite.config.mjs#L9) ([credit](https://stackoverflow.com/a/70936481)).
